cdap-tms/src/main/java/io/cdap/cdap/messaging/store/leveldb/LevelDBTableFactory.java

```diff
@@ -152,8 +153,11 @@ private void upgradeTables(long now) throws IOException {
         DirUtils.deleteDirectoryContents(v2TopicPartitionDir);
       }
 
+      // Make sure appropriate directories are created. This is needed because File.move() call does not create new
+      // directories.
+      Files.createDirectories(v2TopicPartitionDir.toPath());
       LOG.info("Upgrading {} to {}", dirName, v2TopicPartitionDir.getName());
-      Files.move(tableDir.toPath(), v2TopicPartitionDir.toPath());
+      Files.move(tableDir.toPath(), v2TopicPartitionDir.toPath(), StandardCopyOption.ATOMIC_MOVE);
```

Comment:
If Atomic is not possible (e.g. different FS), would it fail?

Reply:
right, in this case its not needed, fixed